search/includes/classes/class-health.php

```diff
@@ -954,17 +954,19 @@ public function get_index_settings_health_for_all_indexables() {
 		return $unhealthy;
 	}
 
-	public function get_active_index_settings_diff_for_indexable( \ElasticPress\Indexable $indexable ) {
-		$version = $this->search->versioning->get_active_version_number( $indexable );
-
-		$version_result = $this->get_index_settings_diff_for_indexable( $indexable, array(
-			'index_version' => $version,
-		) );
+	public function get_index_versions_settings_diff_for_indexable( \ElasticPress\Indexable $indexable ) {
+		$versions = $this->search->versioning->get_versions( $indexable );
 
 		$diff = [];
 
-		if ( ! empty( $version_result ) ) {
-			$diff[] = $version_result;
+		foreach ( $versions as $version ) {
+			$version_result = $this->get_index_settings_diff_for_indexable( $indexable, array(
+				'index_version' => $version['number'],
+			) );
+
+			if ( ! empty( $version_result ) ) {
```

Comment:
I would prefer to use something more explicit than `empty` and I admit it is entirely personal preference.

Consider the merit of this code:

```php
if ( is_array( $version_result ) && ( 0 < count( $version_result ) ) {
```

I think it's less terse, but it's more direct, and it doesn't carry any of the pitfalls that can happen if and when `$version_result` ends up as the wrong type for some reason. I like the preciseness of the statement that this code could only execute if its an array and if it has a length of greater than zero.

I looked into the function `get_index_settings_diff_for_indexable` and it looks like this can only return an empty array in the form of: `[]` or a dictionary type array with the specific object graph that this function is meant to produce.

This is a subtle nuance of PHP being kind of gross and wanting to call a potential list the same thing as a potential dictionary.

Reply:
Thanks for the feedback. It does look like it can also be a WP_Error, so we definitely don't want that to bubble up. I've addressed your feedback now 🙂